### PR TITLE
API: enable pytools Expression representations for all DF estimators

### PR DESCRIPTION
**Pull request by j-ittner**
With this PR, class `EstimatorDF` now implements the `HasExpressionRepr` mix-in, rendering estimator representations as `Expression` objects to enable better formatting.

### CODE REVIEW

File: src/sklearndf/_sklearndf.py [hunk 1]
```diff
@@ -148,6 +161,69 @@ def _get_n_outputs(self) -> int:
         # get the number of outputs this estimator has been fitted to
         pass
 
+    def to_expression(self) -> Expression:
+        """[see superclass]"""
+
+        estimator_parameters = {
```
Comment (mtsokol) [inline src/sklearndf/_sklearndf.py:169]: If we keep default values here would make sense renaming to `estimator_default_parameters` or `default_parameters`?

File: src/sklearndf/_sklearndf.py [hunk 2]
```diff
@@ -148,6 +161,69 @@ def _get_n_outputs(self) -> int:
...
+                    default_value != inspect.Signature.empty
+                )
+                and (
+                    # if the value is an estimator ...
+                    not isinstance(value, BaseEstimator)
+                    # ... then it has the same class as the default value
+                    or value.__class__ == default_value.__class__
+                )
+                and (
+                    # both value and default value are both are np.nan ...
```
Comment (mtsokol) [inline src/sklearndf/_sklearndf.py]: (nit-pick) Would removing repetition be OK? To:
```
# both value and default value are np.nan ...
```

Comment (j-ittner) [reply src/sklearndf/_sklearndf.py]: Good catch! Will update.

File: src/sklearndf/_sklearndf.py (hunk 1 above)
Comment (j-ittner) [reply src/sklearndf/_sklearndf.py:169]: I would keep the name as is, it's a dict with all parameters, whether they have default values or not.

Comment (mtsokol) [reply src/sklearndf/_sklearndf.py:169]: Sure! Then for me it's good to go.

**Review by mtsokol — APPROVED** [review]
Looks good to me! 
(I reviewed it as far as I can, I also looked at `pytools.expression` implementation to get as much understanding of its mechanism as I could for these changes)

**Pull request merged**